components/Web3Provider.tsx

```diff
@@ -216,6 +221,34 @@ export const Web3Provider = ({ children }: Web3ProviderProps): JSX.Element => {
 		setCurrentUser(newUserData)
 	}
 
+	/**
+	 * Creates an identity that the user can identify with
+	 * @param provider - Provider should be either 'Twitter' 'Github' or 'Reddit'
+	 * @returns identity - The identity of that user
+	 */
+	const createIdentity = async (provider: string) => {
+		const ethereumProvider = (await detectEthereumProvider()) as any
+		const ethersProvider = new providers.Web3Provider(ethereumProvider)
+		const signer = ethersProvider.getSigner()
```

Comment:
Since `signer` is ultimately made available as part of the initial `loadWeb3()`, these calls have already happened at this point and are redundant. `signer` should likely be stored as a local state variable so that it may be reused here within `createIdentity`.